File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    # 약수의 개수를 구한다
    # 정수의 제곱이면 홀수 아니면 짝수임
    answer = 0
    for i in range(left, right+1):
        if (i ** 0.5) % 1 == 0:
            answer -= i
        else:
            answer += i
    return answer

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
const solution = (left, right) => {
    let result = 0;
    for (let i = left; i <= right; i++) {
        if (Math.sqrt(i) % 1 === 0) {
            result -= i
        } else {
            result += i
        }
    }
    return result
}